### ISSUE TITLE
feature flags / "experimental" features

### ISSUE THREAD

**Issue by meejah** [LeastAuthority/magic-folder#680]
magic-folder currently exposes a local HTTP API starting with `/v1/...`.

This allows integrating UIs or other programs to depend on functionality.

However, it does not allow for magic-folder to experiment with ideas or "not ready to be supported" APIs.

It would also be good for the CLI to have a way to allow users to "opt in" to features that are in a state that allows useful experimentation but that are _not_ in a fully-baked state.

In short: "feature flags".

What this means precisely here is still somewhat open, but some points:

- local only, no "phone home" (lots of "feature flag" libraries want to contact some web service to ask what features they're "allowed" or "supposed to" enable, but this is not that).
- still useful: "not fully baked" still doesn't mean it's appropriate to push the weekend's hacking to main -- that should still happen on branches etc.

**Comment by meejah** [comment]
Now implemented.
